From 5e8a2b3c4d5e6f708192a3b4c5d6e7f8091a2b3c Mon Sep 17 00:00:00 2001
From: Otavio Morais <otaviomorais@github.com>
Date: Mon, 10 Aug 2026 00:00:00 +0000
Subject: [PATCH] cgroup: netprio: use css->id (cgroup->id removed by 5.x graft)

The 5.x cgroup header graft dropped struct cgroup::id in favor of
cgrp->kn->id, which breaks the 4.19 netprio_cgroup.c (it reads
css->cgroup->id in 4 places). Mainline v5.6 already uses css->id here,
and the grafted cgroup-defs.h still has css->id, so the 4.19 code
compiles unmodified except for this accessor. Enables
CONFIG_CGROUP_NET_PRIO on the base.

Signed-off-by: Otavio Morais <otaviomorais@github.com>
---
 net/core/netprio_cgroup.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/net/core/netprio_cgroup.c b/net/core/netprio_cgroup.c
index 239786608ee4..7b1929498c3f 100644
--- a/net/core/netprio_cgroup.c
+++ b/net/core/netprio_cgroup.c
@@ -97,7 +97,7 @@ static int extend_netdev_table(struct net_device *dev, u32 target_idx)
 static u32 netprio_prio(struct cgroup_subsys_state *css, struct net_device *dev)
 {
 	struct netprio_map *map = rcu_dereference_rtnl(dev->priomap);
-	int id = css->cgroup->id;
+	int id = css->id;
 
 	if (map && id < map->priomap_len)
 		return map->priomap[id];
@@ -117,7 +117,7 @@ static int netprio_set_prio(struct cgroup_subsys_state *css,
 			    struct net_device *dev, u32 prio)
 {
 	struct netprio_map *map;
-	int id = css->cgroup->id;
+	int id = css->id;
 	int ret;
 
 	/* avoid extending priomap for zero writes */
@@ -181,7 +181,7 @@ static void cgrp_css_free(struct cgroup_subsys_state *css)
 
 static u64 read_prioidx(struct cgroup_subsys_state *css, struct cftype *cft)
 {
-	return css->cgroup->id;
+	return css->id;
 }
 
 static int read_priomap(struct seq_file *sf, void *v)
@@ -243,7 +243,7 @@ static void net_prio_attach(struct cgroup_taskset *tset)
 	cgroup_sk_alloc_disable();
 
 	cgroup_taskset_for_each(p, css, tset) {
-		void *v = (void *)(unsigned long)css->cgroup->id;
+		void *v = (void *)(unsigned long)css->id;
 
 		task_lock(p);
 		iterate_fd(p->files, 0, update_netprio, v);
